math_pendulum.cpp

```diff
@@ -2,12 +2,28 @@
 #include "vector.hpp"
 #include "json.hpp"
 #include "simulation_methods.hpp"
+#include <cmath>
 using json = nlohmann::json;
 
 
-double func(double t, Vector y, json conf){
+void func(double t, Vector* y, Vector* y_d, json* config){
+        int n = (*config)["order"].get<int>();
+        double l = (*config)["l"].get<double>();
```

Comment:
I'd suggest extracting values from config with some helper (say parse_config or validate_config) to access values from config just like constant variables or fild of a constant struct.